### ISSUE TITLE
Investigate usage and relationship between 

### ISSUE THREAD

**Issue by lorenzo-dev1** [autonity/autonity#944]
`core/tx_pool.go` has a `GasPrice` field (https://github.com/autonity/autonity/blob/8b4a17c18951088c3af01c06da707a87d7887971/core/tx_pool.go#L238) that is used to decide whether to allow txs into the pool based on their tip (https://github.com/autonity/autonity/blob/8b4a17c18951088c3af01c06da707a87d7887971/core/tx_pool.go#L629). It is set by default to 1 wei.

We have also another `GasPrice` field in the miner config (https://github.com/autonity/autonity/blob/8b4a17c18951088c3af01c06da707a87d7887971/miner/miner.go#L58). This one is set by default to 500_000_000 weis.

The miner `GasPrice` is used for the gas price oracle and to set the gasprice in the eth object. Moreover sometimes it is used to overwrite the txpool gasPrice value. For example:
-  when forcing the mining start trough api or `--mine` flag https://github.com/autonity/autonity/blob/8b4a17c18951088c3af01c06da707a87d7887971/eth/backend.go#L489
- when setting the gas price through the API https://github.com/autonity/autonity/blob/8b4a17c18951088c3af01c06da707a87d7887971/eth/api.go#L128

TL;DR We have two separated fields (one in miner and one in txpool) that both represent the minimum tip we want to accept in our txpool (and consequently in our blocks). Based on different states the one in miner might overwrite the one in txpool. 

I think we should at least have uniform default values for those two parameters. Moreover the way they are handled is confusing, I do not really understand why do we have two parameters to begin with.
